src/silx/opencl/atomic.py

```diff
@@ -42,7 +42,7 @@
 def check_atomic32(device):
     try:
         ctx = pyopencl.Context(devices=[device])
-    except:
+    except Exception:
```

Comment:
Note: bare except captures CTRL-C for example which you don't want for application.

Capture `BaseException` if you really need a cleanup regardless of the exception type. But always re-raise after. Library code should never absorb base exceptions like `KeyboardInterrupt` and `SystemError`.

I'm sure you know but just in case/